### DISCUSSION TITLE
关于 usePropsValue 的讨论

### DISCUSSION THREAD

**Discussion by 19Qingfeng** [ant-design/ant-design-mobile#6266] [General]
关于 `src/utils/use-props-value.ts` 有些疑惑。

我看 antd 和 rc 中关于受控/非受控的实现是使用 [`useMergedState`](https://github.com/react-component/util/blob/master/src/hooks/useMergedState.ts)。

就两种代码的实现方式来说，显然我认为 rc 中的方式比较 antd-mobile 中的更加优秀，这里并不是说代码风格和功能设计上。

而是针对受控组件这里显然会多一次 update 的渲染：

<img width="650" alt="image" src="https://github.com/ant-design/ant-design-mobile/assets/31119562/5977a84b-0ec4-4786-9652-76295d5ad970">

我知道这个问题并不是 Bug 同时也看起来有点吹毛求疵，不过还是想知道关于社区如何看待这种做法。或许我们可以将这种方式进行优化？我愿意为此提出一个 PR。

**Comment by afc163** [comment]
两边可以逐步统一掉一些 hooks 和工具方法的使用。

**Comment by miracles1919** [comment]
当然欢迎 PR 啦~
